libtcmu.c

```diff
@@ -277,78 +268,43 @@ static int generic_handle_cmd(struct tcmu_device *dev,
 	case MODE_SELECT:
 	case MODE_SELECT_10:
 		return tcmu_emulate_mode_select(cdb, iovec, iov_cnt, sense);
+	default:
+		tcmu_err("unknown command %x\n", cdb[0]);
+		return TCMU_NOT_HANDLED;
+	}
+}
+
+static bool command_is_store_call(uint8_t cmd)
+{
```

Comment:
For the question/example about something like compare_and_write, you can make it a store callout, and handlers that do not have a native implementation can just set the compare_and_write callout to a lib function that uses read/write or if the handler's compare_and_write callout is not set just drop down the lib function.

Even if only one handler implements a handler specific compare_and_write callout, all handlers would still get the benefit of COMPARE_AND_WRITE preprocessing and error handling. For example, the SCSI offset and length and iov checks would be done in the generic code, and so would translation of errors to SAM codes.